crates/build/src/lib.rs

```diff
@@ -808,6 +815,37 @@ pub fn execute(args: ExecuteArgs) -> Result<BuildResult> {
     })
 }
 
+/// Modifies permissions of all entries in the dir to be read_only = false
+#[allow(clippy::permissions_set_readonly_false)]
+fn change_permissions(p: &PathBuf) -> Result<()> {
```

Comment:
Maybe try the approach of running the container as the current user e.g. https://unix.stackexchange.com/a/627028

Reply:
I thought you had it figured out @SkymanOne (you said so) but it seems like it's still unfinished.

Give a try to [this approach](https://github.com/Cardinal-Cryptography/ink-wrapper/blob/main/Makefile#L77) - it should run the docker container as the current user.

Reply:
Wouldn't work on windows

Reply:
> I thought you had it figured out @SkymanOne but it seems like it's still causing some troubles. Not surprising, we all were there.
> 
> Give a try to [this approach](https://github.com/Cardinal-Cryptography/ink-wrapper/blob/main/Makefile#L77) - it should run the docker container as the current user.

The problem with that is that it requires rebuilding an image for every newly spawned container. While some intermediate layers may be cached, there is still a potential issue with determinism. Additionally, as I mentioned above, this method will not work on Windows machines.

So, the only solution I see currently is to set permission's read_only flag to false which would give access to the `target` folder to all users

Reply:
> The problem with that is that it requires rebuilding an image for every newly spawned container.

I'm displaying my lack of docker knowledge here, are you saying that specifying the user causes the image to rebuilt each time? I wasn't able to see a difference on this between setting the `--user` and leaving it out and defaulting to `root`

I've played with it a bit and was able to get this to work just using the raw docker commands:

![image](https://github.com/paritytech/cargo-contract/assets/75586/da082f9b-9c3b-4ffd-84c9-9187083f0752)
![image](https://github.com/paritytech/cargo-contract/assets/75586/3f054436-38d8-44e5-ab7e-387def511f49)

> this method will not work on Windows machines

There must be another way though to control the user in windows? In any case the current solution of `@SkymanOne
[give permission to target to everyone](https://github.com/paritytech/cargo-contract/pull/1148/commits/217ab6a49338e35ba7ba19b4fb4ca6cb970db10c)` does not work on Linux for me at the moment either.

There must be a way we can get this working on all OSs.

Reply:
Managed to fix the file permission issue on linux. Now just need to figure out how to get the conditional compilation code working

Reply:
Permissions issue fixed now :+1: